crates/exex/src/atlantic/model.rs

```diff
@@ -0,0 +1,160 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct Context {
+    #[serde(rename = "proofPath")]
+    pub proof_path: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct Job {
+    pub id: String,
+    pub status: String,
+    pub context: Option<Context>,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct SharpQueryDetails {
+    id: String,
+    submitted_by_client: String,
+    status: String,
+    step: Option<String>,
+    program_hash: Option<String>,
+    layout: Option<String>,
+    program_fact_hash: Option<String>,
+    price: String,
+    credits_used: usize,
+    is_fact_mocked: bool,
+    prover: Option<String>,
+    chain: String,
+    steps: Vec<String>,
+    created_at: String,
+    completed_at: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct QueryResponse {
+    pub sharp_query_id: String,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json;
+
+    #[test]
+    fn test_context_serialization() {
+        let context = Context { proof_path: Some("path/to/proof".to_string()) };
+
+        let serialized = serde_json::to_string(&context).expect("Failed to serialize Context");
+        assert_eq!(
+            serialized, r#"{"proofPath":"path/to/proof"}"#,
+            "Serialization of Context is incorrect"
+        );
+
+        let deserialized: Context =
+            serde_json::from_str(&serialized).expect("Failed to deserialize Context");
+        assert_eq!(deserialized, context);
```

Comment:
this part of the test is a bit odd because you test that `serde` is consistent with itself, precisely you do `serde_json::from_str(serde_json::to_string(content)) == content`